### ISSUE TITLE
The Context Menu text of TextField truncated on Flutter v13.3.5 when using larger text on iOS 16.7

### ISSUE THREAD

**Issue by lmshek** [flutter/flutter#136527]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

I have updated the Flutter version recently from v3.7.1 to v3.13.5. I have found that the context menu text has been truncated in the text editor on v3.13.5 but not in v3.7.1 on iOS 16.7.

Step to reproduce:
1. Enlarge the text on iOS by Settings -> Accessibility -> Display & Text Size -> Larger Text
2. Using fvm to install Flutter v3.13.5
3. Run the Flutter Project with TextField in the code.
4. Copy some text in other app.
5. Go back to the Flutter app and long press the TextField.
6. The "Paste" button in the context menu is truncated. (You can find the screenshots in the "Screenshots or Video" section)

### Expected results

The text of the context menu should not be truncated even I have enlarged the OS font size.

### Actual results

The text of the context menu has been truncated if I have enlarged the OS font size.

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Flutter v3.7.1](https://drive.google.com/file/d/1OZL4TKVIY-fUTN5V6NN2fdCLQwTfOJ5B/view?usp=drive_link)
[Flutter v3.13.5](https://drive.google.com/file/d/1_87AOzvxDVpXzUiGq82m-G_2Z6mo0MmO/view?usp=drive_link)

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.5, on macOS 13.5.2 22G91 darwin-arm64, locale en-HK)
    • Flutter version 3.13.5 on channel stable at /Users/lmshek/fvm/versions/3.13.5
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 12fccda598 (3 weeks ago), 2023-09-19 13:56:11 -0700
    • Engine revision bd986c5ed2
    • Dart version 3.1.2
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/lmshek/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = /Users/lmshek/Library/Android/sdk
    • Java binary at: /Applications/Android [Studio.app/Contents/jre/Contents/Home/bin/java](http://studio.app/Contents/jre/Contents/Home/bin/java)
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google [Chrome.app/Contents/MacOS/Google](http://chrome.app/Contents/MacOS/Google) Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android [Studio.app/Contents](http://studio.app/Contents)
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.70.2)
    • VS Code at /Users/lmshek/Documents/Visual Studio [Code.app/Contents](http://code.app/Contents)
    • Flutter extension version 3.60.0

[✓] Connected device (4 available)
    • Lok Man’s iPhone (mobile) • 00008030-001D18A23E46802E            • ios            • iOS 16.7 20H19
    • iPhone 12 Pro (mobile)    • D60CF662-001D-4588-93D4-98805AA488B7 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)           • macos                                • darwin-arm64   • macOS 13.5.2 22G91 darwin-arm64
    • Chrome (web)              • chrome                               • web-javascript • Google Chrome 117.0.5938.149

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by dam-ease** [comment]
Hi @lmshek. Thanks for filing this. I can reproduce this on the latest `master` and `stable` channels as well following the steps highlighted above.

| With Normal Text | Native (large Text) | Flutter App (Large Text) |
|--------|--------|--------|
| <img height="500" src="https://github.com/flutter/flutter/assets/53122008/68b22e19-8af0-429e-8055-a5b4c0b68de1"> | <img height="500" src="https://github.com/flutter/flutter/assets/53122008/21224b74-1a80-479d-8faf-a27163ef72c0">  | <img height="500" src="https://github.com/flutter/flutter/assets/53122008/9a74184c-0a39-410c-8852-7da5ac74a34a"> | 

<details><summary>stable, master flutter doctor -v</summary>
<p>

```
[✓] Flutter (Channel stable, 3.13.7, on macOS 13.0 22A380 darwin-arm64, locale
    en-NG)
    • Flutter version 3.13.7 on channel stable at
      /Users/damilolaalimi/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2f708eb839 (3 days ago), 2023-10-09 09:58:08 -0500
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio
      Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build
      17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.82.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (4 available)
    • sdk gphone64 arm64 (mobile) • emulator-5556                        •
      android-arm64  • Android 14 (API 34) (emulator)
    • iPhone XS (mobile)          • 3CFD1268-0D52-486E-A12E-6E136E972D94 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)             • macos                                •
      darwin-arm64   • macOS 13.0 22A380 darwin-arm64
    • Chrome (web)                • chrome                               •
      web-javascript • Google Chrome 117.0.5938.149

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
``` 
```
[✓] Flutter (Channel master, 3.16.0-10.0.pre.72, on macOS 13.0 22A380 darwin-arm64, locale en-NG)
    • Flutter version 3.16.0-10.0.pre.72 on channel master at /Users/damilolaalimi/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d479b5e2c6 (72 minutes ago), 2023-10-13 03:35:07 -0400
    • Engine revision b59d779d4f
    • Dart version 3.3.0 (build 3.3.0-20.0.dev)
    • DevTools version 2.28.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.82.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (4 available)
    • sdk gphone64 arm64 (mobile) • emulator-5556                        • android-arm64  • Android 14 (API 34) (emulator)
    • iPhone XS (mobile)          • 3CFD1268-0D52-486E-A12E-6E136E972D94 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)             • macos                                • darwin-arm64   • macOS 13.0 22A380 darwin-arm64
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome 117.0.5938.149

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
``` 

</p>
</details>

**Comment by LongCatIsLooong** [comment]
I believe this has already been fix by https://github.com/flutter/flutter/pull/133386. The earliest version with the fix is [3.15.0-6.0.pre](https://github.com/flutter/flutter/releases/tag/3.15.0-6.0.pre)

**Comment by dam-ease** [comment]
Thanks for your input @LongCatIsLooong. Tested this on the latest `master` channel and I can see this doesn't reproduce there.
Closing as this has been fixed in the latest versions.
